File: hospital-Management-System-Backend/hospital-Management-System-Backend/src/main/java/com/project/hospital_Management_System_Backend/dtos/FullPatientDto.java
package com.project.hospital_Management_System_Backend.dtos;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@NoArgsConstructor
@Data
public class FullPatientDto {
    private Long id;

    private ProfileDto profile;

    private int age;

    private String gender;

    private String address;

    private String prognosis;

    private Set<CompanyDto> companies;

    private Set<BasicEmployeeDto> employees;

    private Set<AppointmentsDto> appointments;

    private Set<PrescriptionsDto> prescriptions;

    private Set<ProceduresDto> procedures;
}

File: frontend/hospital-Management-System-frontend/src/app/models/FullPatientDTO..ts
import AppointmentDTO from "./AppointmentDTO";
import BasicEmployeeDTO from "./BasicEmployeeDTO";
import CompanyDTO from "./CompanyDTO";
import PrescriptionDTO from "./PrescriptionDTO";
import ProcedureDTO from "./ProcedureDTO";
import ProfileDTO from "./ProfileDTO";

export default interface FullPatientDTO{
    id:number,
    profile:ProfileDTO,
    age:number,
    gender:string,
    address:string,
    prognosis:string,
    companies:CompanyDTO[],
    employees:BasicEmployeeDTO[],
    appointments:AppointmentDTO[],
    prescriptions:PrescriptionDTO[]
    procedures:ProcedureDTO[]
}